module/VuFind/src/VuFind/ILS/Driver/GeniePlus.php

```diff
@@ -0,0 +1,584 @@
+    {
+        $bibId = current(
+            $this->extractDisplayValues(
+                $this->getFieldFromApiRecord($record, 'id')
+            )
+        );
+        $barcodes = $this->extractDisplayValues(
+            $this->getFieldFromApiRecord($record, 'barcode')
+        );
+        $callNos = $this->extractDisplayValues(
+            $this->getFieldFromApiRecord($record, 'callnumber')
+        );
+        $dueDates = $this->extractDisplayValues(
+            $this->getFieldFromApiRecord($record, 'duedate')
+        );
+        $locations = $this->extractDisplayValues(
+            $this->getFieldFromApiRecord($record, 'location')
+        );
+        $statuses = $this->extractDisplayValues(
+            $this->getFieldFromApiRecord($record, 'status')
+        );
+        $volumes = $this->extractDisplayValues(
+            $this->getFieldFromApiRecord($record, 'volume')
+        );
+        $total = max(
+            [
+                count($barcodes),
+                count($callNos),
+                count($dueDates),
+                count($locations),
+                count($statuses),
+                count($volumes),
+            ]
+        );
+        $result = [];
+        for ($i = 0; $i < $total; $i++) {
+            $availability = in_array($statuses[$i] ?? '', $this->availableStatuses)
+                ? 1 : 0;
+            $result[] = [
+                'id' => $bibId,
+                'availability' => $availability,
+                'status' => $statuses[$i] ?? '',
+                'location' => $locations[$i] ?? '',
+                'reserve' => 'N', // not supported
+                'callnumber' => $callNos[$i] ?? '',
+                'duedate' => $dueDates[$i] ?? '',
+                'number' => $volumes[$i] ?? ($i + 1),
+                'barcode' => $barcodes[$i] ?? '',
+            ];
+        }
+        return $result;
+    }
+
+    /**
+     * Get the search path to query a template.
+     *
+     * @param string $template Name of template to query
+     *
+     * @return string
+     */
+    protected function getTemplateQueryPath(string $template): string
+    {
+        $database = $this->config['API']['database'];
+        return "/_rest/databases/$database/templates/$template/search-result";
+    }
+
+    /**
+     * Get Status
+     *
+     * This is responsible for retrieving the status information of a certain
+     * record.
+     *
+     * @param string $id The record id to retrieve the holdings for
+     *
+     * @return mixed     On success, an associative array with the following keys:
+     * id, availability (boolean), status, location, reserve, callnumber.
+     */
+    public function getStatus($id)
+    {
+        $template = $this->config['API']['catalog_template'];
+        $path = $this->getTemplateQueryPath($template);
+        $idField = $this->config['Item']['field']['id'];
+        $safeId = str_replace("'", '', $id); // don't allow quotes in IDs
```

Comment:
A helper method that returns a safe id or a safe search string would be nice, since this is done in more than one place.